xbmc/addons/ContextMenuAddon.cpp

```diff
@@ -54,26 +54,26 @@ void CContextMenuAddon::ParseMenu(
 
   items.push_back(CContextMenuItem::CreateGroup(menuLabel, parent, menuId, addonInfo.ID()));
 
-  ELEMENTS subMenus;
-  if (CServiceBroker::GetAddonMgr().GetExtElements(elem, "menu", subMenus))
-    for (const auto& subMenu : subMenus)
-      ParseMenu(addonInfo, subMenu, menuId, anonGroupCount, items);
-
-  ELEMENTS elems;
-  if (CServiceBroker::GetAddonMgr().GetExtElements(elem, "item", elems))
+  for (unsigned int i = 0; i < elem->num_children; i++)
   {
-    for (const auto& elem : elems)
+    cp_cfg_element_t subElem = elem->children[i];
+    std::string elementName = subElem.name;
+    if (elementName.compare("menu") == 0)
+      ParseMenu(addonInfo, &subElem, menuId, anonGroupCount, items);
+
+    else if (elementName.compare("item") == 0)
     {
-      auto visCondition = CServiceBroker::GetAddonMgr().GetExtValue(elem, "visible");
-      auto library = CServiceBroker::GetAddonMgr().GetExtValue(elem, "@library");
-      auto label = CServiceBroker::GetAddonMgr().GetExtValue(elem, "label");
+      auto visCondition = CServiceBroker::GetAddonMgr().GetExtValue(&subElem, "visible");
+      auto library = CServiceBroker::GetAddonMgr().GetExtValue(&subElem, "@library");
+      auto label = CServiceBroker::GetAddonMgr().GetExtValue(&subElem, "label");
       if (StringUtils::IsNaturalNumber(label))
         label = g_localizeStrings.GetAddonString(addonInfo.ID(), atoi(label.c_str()));
```

Comment:
Could use `std::stoi`

Reply:
This code was already there, I just moved it from one loop into another. But I will have a look.